### PR TITLE
fix(gatsby): fix root resolving of resources

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

Fixes url resolving by ignoring / queries but allow relative ones to be resolved by webpack.

Currently this doesn't load in the browser
```css
background-image: url('./image.png');
```

This does when it's inside the /static folder or copied to the public folder.
```css
background-image: url('/image.png');
```

With this PR both work.

I've also added a test to see that we don't regress in the future.

### CODE REVIEW

File: e2e-tests/production-runtime/cypress/integration/1-production.js [hunk 1]
```diff
@@ -132,7 +132,7 @@ describe(`Production build tests`, () => {
 
   describe(`Supports unicode characters in urls`, () => {
     it(`Can navigate directly`, () => {
-      cy.visit(`/안녕/`, {
+      cy.visit(encodeURI(`/안녕/`), {
```
Comment (wardpeet) [inline e2e-tests/production-runtime/cypress/integration/1-production.js:135]: necessary for cypress upgrade

File: packages/gatsby/src/utils/webpack-utils.ts [hunk 2]
```diff
@@ -279,7 +279,14 @@ export const createWebpackUtils = (
       return {
         loader: require.resolve(`css-loader`),
         options: {
-          url: false,
+          url: function (url: string): boolean {
+            // When an url starts with /
+            if (url.startsWith(`/`)) {
```
Comment (vladar) [inline packages/gatsby/src/utils/webpack-utils.ts:284]: What happens when someone uses fully-qualified URLs? I.e. `url("https://example.com/some-image.jpg")`

Comment (wardpeet) [reply packages/gatsby/src/utils/webpack-utils.ts:284]: I did not test :) good one!

File: e2e-tests/production-runtime/cypress/integration/resource-loading-resilience.js [hunk 3]
```diff
@@ -63,52 +63,46 @@ const runTests = (testNameSuffix = `Unknown scenario`) => {
...
+  before(() => {
+    cy.task("getAssetsForPage", {
+      pagePath: args.pagePath,
+      filter: args.filter,
+    }).then(urls => {
+      for (const url of urls) {
+        cy.intercept(url, {
+          statusCode: 404,
+          body: "",
         })
```
Comment (pieh) [inline e2e-tests/production-runtime/cypress/integration/resource-loading-resilience.js:76]: does those stay blocked for following tests? Do we need to "unintercept" those so next steps don't "inherit" setup with resources blocked from previous runs?

File: packages/gatsby/src/utils/webpack-utils.ts (hunk 2 above)
Comment (pieh) [reply packages/gatsby/src/utils/webpack-utils.ts:284]: And similar case (just without protocol): `url("//example.com/some-image.jpg")`

Comment (vladar) [reply packages/gatsby/src/utils/webpack-utils.ts:284]: This one starts with `/` 😄

Comment (pieh) [reply packages/gatsby/src/utils/webpack-utils.ts:284]: 🤦

File: packages/gatsby/src/utils/webpack-utils.ts [hunk 4]
```diff
@@ -279,7 +279,14 @@ export const createWebpackUtils = (
       return {
         loader: require.resolve(`css-loader`),
         options: {
-          url: false,
+          url: function (url: string): boolean {
```
Comment (wardpeet) [inline packages/gatsby/src/utils/webpack-utils.ts:282]:
```suggestion
          // Absolute urls (https or //) are not send to this function. Only resolvable paths absolute or relative ones.
          url: function (url: string): boolean {
```

**Review by pieh — APPROVED** [review]

**Pull request merged**